imgui-lwjgl3/src/main/java/imgui/glfw/ImGuiImplGlfw.java

```diff
@@ -0,0 +1,405 @@
+        mapButton(ImGuiNavInput.DpadLeft, 13, buttons, buttonsCount, io);    // D-Pad Left
+        mapButton(ImGuiNavInput.DpadRight, 11, buttons, buttonsCount, io);    // D-Pad Right
+        mapButton(ImGuiNavInput.DpadUp, 10, buttons, buttonsCount, io);    // D-Pad Up
+        mapButton(ImGuiNavInput.DpadDown, 12, buttons, buttonsCount, io);    // D-Pad Down
+        mapButton(ImGuiNavInput.FocusPrev, 4, buttons, buttonsCount, io);     // L1 / LB
+        mapButton(ImGuiNavInput.FocusNext, 5, buttons, buttonsCount, io);     // R1 / RB
+        mapButton(ImGuiNavInput.TweakSlow, 4, buttons, buttonsCount, io);     // L1 / LB
+        mapButton(ImGuiNavInput.TweakFast, 5, buttons, buttonsCount, io);     // R1 / RB
+        mapAnalog(ImGuiNavInput.LStickLeft, 0, -0.3f, -0.9f, axis, axisCount, io);
+        mapAnalog(ImGuiNavInput.LStickRight, 0, +0.3f, +0.9f, axis, axisCount, io);
+        mapAnalog(ImGuiNavInput.LStickUp, 1, +0.3f, +0.9f, axis, axisCount, io);
+        mapAnalog(ImGuiNavInput.LStickDown, 1, -0.3f, -0.9f, axis, axisCount, io);
+
+        if (axisCount > 0 && buttonsCount > 0) {
+            io.setBackendFlags(io.getBackendFlags() | ImGuiBackendFlags.HasGamepad);
+        } else {
+            io.setBackendFlags(io.getBackendFlags() & ~ImGuiBackendFlags.HasGamepad);
+        }
+    }
+
+    private void mapButton(final int navNo, final int buttonNo, final ByteBuffer buttons, final int buttonsCount, final ImGuiIO io) {
+        if (buttonsCount > buttonNo && buttons.get(buttonNo) == GLFW_PRESS) {
+            io.setNavInputs(navNo, 1.0f);
+        }
+    }
+
+    private void mapAnalog(
+        final int navNo,
+        final int axisNo,
+        final float v0,
+        final float v1,
+        final FloatBuffer axis,
+        final int axisCount,
+        final ImGuiIO io
+    ) {
+        float v = axisCount > axisNo ? axis.get(axisNo) : v0;
+        v = (v - v0) / (v1 - v0);
+        if (v > 1.0f) {
+            v = 1.0f;
+        }
+        if (io.getNavInputs(navNo) < v) {
+            io.setNavInputs(navNo, v);
+        }
+    }
+
+    /**
+     * Updates {@link ImGuiIO} and {@link org.lwjgl.glfw.GLFW} state.
+     */
+    public void newFrame() {
+        final ImGuiIO io = ImGui.getIO();
+        if (!io.getFonts().isBuilt()) {
+            throw new IllegalStateException(
+                "Font atlas not built! It is generally built by the renderer back-end. Missing call to renderer init() method? e.g. ImGuiImplGl3.init().");
+        }
+
+        final int[] width = new int[1];
+        final int[] height = new int[1];
+        glfwGetWindowSize(windowId, width, height);
+
+        final int[] displayWidth = new int[1];
+        final int[] displayHeight = new int[1];
+        glfwGetFramebufferSize(windowId, displayWidth, displayHeight);
+
+        io.setDisplaySize((float) width[0], (float) height[0]);
+        if ((float) width[0] > 0 && (float) height[0] > 0) {
+            io.setDisplayFramebufferScale(
```

Comment:
I've noticed it only now, but you need to set `DisplaySize`/`DisplayFramebufferScale`/`MousePos` like here

https://github.com/SpaiR/imgui-java/commit/c024de2b2162bc08bc605423f3e304c12de84c0b#diff-f73903ceb653ba7a505e28080dbcfa6e

Otherwise there could be problems with mouse positioning.

Reply:
Hadn't noticed that, thanks for pointing out.